build-tools-internal/src/test/java/org/elasticsearch/gradle/internal/release/SetCompatibleVersionsTaskTests.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.gradle.internal.release;
+
+import com.github.javaparser.StaticJavaParser;
+import com.github.javaparser.ast.CompilationUnit;
+
+import org.junit.Test;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.hasToString;
+
+public class SetCompatibleVersionsTaskTests {
+
+    @Test
+    public void updateMinCcsVersion() {
+        final String transportVersionsJava = """
+            public class TransportVersions {
+                public static final TransportVersion V1 = def(100);
+                public static final TransportVersion V2 = def(200);
+                public static final TransportVersion V3 = def(300);
+
+                public static final TransportVersion MINIMUM_CCS_VERSION = V2;
+            }""";
+        final String updatedJava = """
+            public class TransportVersions {
+
+                public static final TransportVersion V1 = def(100);
```

Comment:
I thought we have a hack to avoid the additional newlines. Is this test not using that code?

Reply:
This isn't using  the lexigraphical parser, just the default one (which uses a different output method). This test is just checking the semantic modification